### PR TITLE
Handle disable IPv6 with old network-manager

### LINKED ISSUE #2364: Host System: IPv6 disabled makes Supervisor unable to save
### Describe the issue

While editing the `Host System` network details in Network settings Supervisor menu (i.e. changing DNS address in IPv4 connection), while having IPv6 Disabled, makes it unable to save the data.

Starting point (after edit of IPv4, leaving IPv6 disabled as it was):
image
after clicking 'Save':
image
(translated: ipv6.method: property is invalid)

### Steps to reproduce

1. Go to Supervisor -> System -> host System
2. Change anything but have IPv6 disabled 
3. Try to save

### Enviroment details

- **Operating System:**: Debian 10 Supervised / no issues reported by Supervisor
- **Supervisor version:**: 2020.12.6
- **Home Assistant version**: 2020.12.0

### Supervisor logs

<details>
<summary>Supervisor logs</summary>

```
20-12-14 01:42:05 ERROR (MainThread) [supervisor.host.network] Can't update config on enp2s0: Error: GDBus.Error:org.freedesktop.NetworkManager.Settings.Connection.InvalidProperty: ipv6.method: właściwość jest nieprawidłowa

```

</details>

### System Information

<details>
<summary>System Information</summary>

```
arch: amd64
channel: stable
docker: 19.03.14
features:
- reboot
- shutdown
- services
- network
- hostname
hassos: null
homeassistant: 2020.12.0
hostname: brix
logging: info
machine: qemux86-64
operating_system: Debian GNU/Linux 10 (buster)
supervisor: 2020.12.6
supported: true
supported_arch:
- amd64
- i386
timezone: Europe/Warsaw


```

</details>

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

Use link-local instead of disable for old NM

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #2364
- This PR is related to issue:
- Link to documentation pull request:
- Link to cli pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: supervisor/host/network.py [hunk 1]
```diff
@@ -121,6 +124,16 @@ async def apply_changes(self, interface: Interface) -> None:
...
+        # Handle issue with disabling IPv6 with old network manager
+        if (
+            interface.ipv6.method == InterfaceMethod.DISABLED
+            and self.sys_dbus.network.version < MIN_DISABLE_IPV6_NM_VERSION
+        ):
+            interface.ipv6.method = InterfaceMethod.LINK_LOCAL
+            _LOGGER.warning(
+                "Method 'disable' could not be used, using 'link-local' instead"
+            )
+
```
Comment (pvizeli) [inline supervisor/host/network.py:136]: Can we move that code into the payload generator?

Comment (ludeeus) [reply supervisor/host/network.py:136]: Yes, but why?

File: supervisor/host/const.py [hunk 2]
```diff
@@ -6,6 +6,7 @@ class InterfaceMethod(str, Enum):
     """Configuration of an interface."""
 
     DISABLED = "disabled"
+    LINK_LOCAL = "link-local"
```
Comment (pvizeli) [inline supervisor/host/const.py:9]: we do not support link-local in our abstraction, but we support it on dbus.

File: supervisor/dbus/payloads/interface_update.tmpl [hunk 3]
```diff
@@ -42,6 +42,8 @@
             'method': <'auto'>
 {% elif interface.ipv6.method == "disabled" %}
             'method': <'disabled'>
+{% elif interface.ipv6.method == "link-local" %}
```
Comment (pvizeli) [inline supervisor/dbus/payloads/interface_update.tmpl:45]: `{% elif interface.ipv6.method == "disabled" and nmversion > xy %}`

**Review by pvizeli — CHANGES_REQUESTED** [review]
We have an abstraction between us and the OS. Our layer only supports these 3 modes and we need to convert it to the best working OS config without extend our layer

**Pull request closed without merging**